_source/_posts/2025-07-08-express-authentication.md

```diff
@@ -0,0 +1,722 @@
+
+   * **Assignments:** Allow everyone in your organization to access.
+
+6. After creating the app, click the edit button under Client Credentials and enable **Require PKCE as additional verification**.
+
+
+7. Copy the **Client ID**, **Client Secret**, and **Okta Domain** and add them to your `.env` file.
+
+## Set up passwordless login using FIDO2 and WebAuthn with Okta 
+
+Add the **FIDO2 (WebAuthn)** authenticator in Okta to enable passwordless login and follow the [Okta documentation](https://help.okta.com/oie/en-us/content/topics/identity-engine/authenticators/configure-webauthn.htm) for complete enrollment and policy configuration instructions.
+
+>  **Note:** When logging in for the first time, make sure you, as the admin, and all assigned users enroll in FIDO authentication. To simplify this process, you can create a user group containing all team members and assign it to the Web App. Then include the group in your enrollment policy, which you can customize as needed.
+ 
+
+## Building the Express app 
+
+Create an `index.js` file in your project root. It serves as the main entry point for your application. Use it to initialize the Express app, configure session handling, set up [Passport](https://www.passportjs.org/) to use OIDC, and set up the routes. 
+
+```javascript
+import express from 'express';
+import session from 'express-session';
+import passport from 'passport'; 
+import routes from './routes.js';
+import expressLayouts from 'express-ejs-layouts';
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(expressLayouts); 
+app.set('layout', 'layout');
+app.use(express.urlencoded({ extended: false }));
+
+app.use(session({
+  secret: "your-hardcoded-secret",
+  resave: false,
+  saveUninitialized: true,
+}));
+
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.serializeUser(function (user, done) {
+  done(null, user);
+});
+
+passport.deserializeUser(function (obj, done) {
+  done(null, obj);
+});
+
+app.use('/', routes);
+
+app.listen(3000, () => {
+  console.log('Server listening on http://localhost:3000');
+});
+```
+
+### Define team mapping and sample expenses 
+
+Create a `utils.js` file to serve as a data module for your project. This file includes a user-to-team mapping and has dummy expense data for each team, covering all teams configured for testing in your web app.
+
+The application determines the user's team context from the email claim in the ID token and filters the expense list accordingly, so the dashboard displays only that team's data.
+
+To customize the data, open utils.js and update the following objects:
+
+ALL_TEAMS_NAME - an array listing all teams in your organization
+
+userTeamMap - maps each user's email (or "admin" for full access) to a specific team
+
+dummyExpenseData - contains sample expense data for each team
+
+
+```javascript
+export const ALL_TEAMS_NAME = ["finance", "hr", "legal", "marketing", "dev advocacy"];
+
+export const userTeamMap = {
```

Comment:
Very very unfortunately, we cannot use these for license reasons.

I recommend using some fake names and I'd also change the domain to something totally fake.

I bought the domain `task-vantage.com` you can use that (or use a invalid TLD like company.fake)

Reply:
Sure, thanks Alisa. I will change it.